tests/integration/test_fim/test_files/test_audit_buffer/test_audit_buffer_behavoir.py

```diff
@@ -0,0 +1,311 @@
+
+    wazuh_min_version: 4.5.0
+
+    tier: 2
+
+    parameters:
+        - configuration:
+            type: dict
+            brief: Configuration values for ossec.conf.
+        - metadata:
+            type: dict
+            brief: Test case data.
+        - test_folders:
+            type: dict
+            brief: List of folders to be created for monitoring.
+        - set_wazuh_configuration:
+            type: fixture
+            brief: Set ossec.conf configuration.
+        - create_monitored_folders:
+            type: fixture
+            brief: Create a given list of folders when the test starts. Delete the folders at the end of the test.
+        - configure_local_internal_options_function:
+            type: fixture
+            brief: Set local_internal_options.conf file.
+        - restart_syscheck_function:
+            type: fixture
+            brief: restart syscheckd daemon, and truncate the ossec.log.
+        - wait_syscheck_start:
+            type: fixture
+            brief: check that the starting FIM scan is detected.
+
+    assertions:
+        - Verify when queue is full an event informs audit events may be lost
+        - Verify that files detected in whodata mode before it being full are equal or more than the configured value
+        - Verify the excess files inserted after queue full are detected in scheduled mode.
+
+    input_description: The file 'configuration_audit_buffer_values' provides the configuration template.
+                       The file 'cases_audit_buffer_values.yaml' provides the tes cases configuration
+                       details for each test case.
+
+    expected_output:
+        - r".*(Internal audit queue is full). Some events may be lost. Next scheduled scan will recover lost data."
+        - r".*Sending FIM event: (.+)$"
+    '''
+    wazuh_log_monitor = FileMonitor(LOG_FILE_PATH)
+    files_to_add = metadata['files_to_add']
+
+    # Insert an ammount of files
+    for file in range(0, files_to_add):
+        create_file(REGULAR, test_folders[0], f'test_file_{file}', content='')
+
+    # Detect If queue_full message has been generated
+    detect_audit_queue_full(wazuh_log_monitor, update_position=False)
+
+    # Get all file added events
+    results = get_messages(callback_detect_file_added_event, timeout=T_10,
+                           error_message=f"Did not receive the expected file added events")
+
+    # Check the ammount of added events in whodata mode is equal or more than the expected value
+    found_whodata_events = 0
+    for result in results:
+        if result['data']['mode'] == 'whodata':
+            found_whodata_events = found_whodata_events + 1
+    assert found_whodata_events >= metadata['whodata_events'], f"Found less whodata File added events \
+                                                                 than the expected {metadata['whodata_events']}"
+
+    # Wait for scheduled scan so the rest of file events are generated
+
```

Comment:
```suggestion
```

Reply:
Done in https://github.com/wazuh/wazuh-qa/pull/3990/commits/2eb80e3b64add4a8e00deb846a1d2a19a8ab4a09